dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/icon/IconControllerTest.java

```diff
@@ -107,30 +104,44 @@ void shouldUpdateIconWhenKeyExists() throws IOException {
     assertEquals(String.format("Icon %s updated", iconKey), response.getString("message").string());
   }
 
+  @Test
+  void shouldGetOnlyCustomIcons() throws IOException {
+    String fileResourceId = createFileResource();
+    createIcon(fileResourceId, keywords);
+
+    JsonArray iconArray = GET("/icons?type=CUSTOM").content(HttpStatus.OK);
+
+    verifyAll(iconArray, keywords, fileResourceId);
+  }
+
+  @Test
+  void shouldGetIconsWithSpecifiedKey() throws IOException {
+    String fileResourceId = createFileResource();
+    createIcon(fileResourceId, keyword);
+
+    JsonArray iconArray = GET("/icons?keywords=m1").content(HttpStatus.OK);
```

Comment:
You can have a look at EnrollmentsExportControllerTest on how we use jsontree 😄 

I think you should be able to do

```suggestion
    JsonList<JsonIcon> icons = GET("/icons?keywords=m1")
       .content(HttpStatus.OK)
       .asList(JsonIcon.class);
```

or `.getList("someKey, JsonIcon.class);` if the icons returned in a map.


All our custom assertions like this are usually prefixed with `assert`. With the above change you can then do

`assertCustomIcon(icons.get(0), ...)`

this makes your assertion reusable as it does not include that you expect the icon to be in position 0 in a list/array.

Reply:
fixed